src/core/geometry/qgsgeometry.cpp

```diff
@@ -2627,7 +2633,12 @@ QgsGeometry QgsGeometry::makeValid() const
     return QgsGeometry();
 
   mLastError.clear();
+#if GEOS_VERSION_MAJOR>3 || ( GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR>=9 )
+  QgsGeos geos( d->geometry.get() );
+  std::unique_ptr< QgsAbstractGeometry > g( geos.makeValid( &mLastError ) );
+#else
```

Comment:
It could be appropriate to add a #if GEOS < 3.8 in qgsgeometrymakevalid.cpp to not build it and make is easy to spot it and remove it in a few years when our baseline is above 3.8

Reply:
Ok, done!